### PR TITLE
optional csv summary output

### PR DESCRIPTION
**Pull request by karlnaden**
# Summary

For systems without robust TestReport functionality, provide the option to generate a simple csv-based summary of the tests execution and whether each passed or failed.

## New behavior

- Added option `summary_path` to config file and cli. Specifying a directory path will enable saving of the execution summary
- During TestReport output, if a `summary_path` is specified, then create a csv file using the current timestamp and a row for each TestScript executed including the id, name, and title of the TestScript and the result (pass/fail)

## Code changes

- execution engine
- cli
- readme

## Testing guidance

- Config file: committed config file writes to the `Summaries` folder. Execute with `bundle exec bin/testscript_engine execute --config config.yml`
- Command line: `bundle exec testscript_engine execute --testscript_name testscript-example-validateprofileid --ext_validator "http://localhost/validatorapi" --summary_path Summaries`

### CODE REVIEW

File: lib/testscript_engine/cli.rb [hunk 1]
```diff
@@ -6,11 +6,12 @@ class TestScriptEngine
 
   module CLI 
     class MyCLI < Thor
-      desc "execute [OPTIONS]", "--config[PATH] --ext_validator [URL] --ext_fhirpath [URL] --variable ['name = value'] --server_url [URL] --testscript_path [FILEPATH] --testreport_path [FILEPATH] --nonfhir_fixture --verbose"
+      desc "execute [OPTIONS]", "--config[PATH] --ext_validator [URL] --ext_fhirpath [URL] --variable ['name = value'] --server_url [URL] --testscript_path [FILEPATH] --testreport_path [FILEPATH] summary_path [FILEPATH] --nonfhir_fixture --verbose"
```
Comment (jhlee-mitre) [inline lib/testscript_engine/cli.rb:9]: Minor thing: [FILEPATH] [PATH] [FILE] been used mixed way and should be clear. --config needs a specific file name whereas --summary_path or --testreport need a path. My vote is to use [FILE] for file and use [PATH] for path. What do you think?

**Review by jhlee-mitre — APPROVED** [review]
Work well and no concerns. Approved.

File: lib/testscript_engine/cli.rb (hunk 1 above)
Comment (karlnaden) [reply lib/testscript_engine/cli.rb:9]: I'm onboard with [PATH]. I'll change just the summary_path stuff with this pull request and leave the others for you to change as a part of other changes you're making to the path handling. Or do you want me to roll it all into this PR?

Comment (karlnaden) [reply lib/testscript_engine/cli.rb]: fixed all path stuff

**Pull request merged**